### DISCUSSION TITLE
Folia Support for  QuickShop-Hikari？

### DISCUSSION THREAD

**Discussion by Hanmingxuan** [QuickShop-Community/QuickShop-Hikari#964] [Ideas]
Here is the repo link: https://github.com/PaperMC/Folia

**Comment by Ghost-chu** [comment]
No, we didn't support Folia yet.

---

抱歉，我们暂不支持 Folia。Folia对调度器进行了大量修改，异步 ticking 使得我们无法在同一个 tick 里完成交易，这会导致交易不一致的问题，并可能引发刷物品的问题。
在Folia对调度器做出进一步改变之前，我们没有计划添加对 Folia 的支持。

**Comment by Ghost-chu** [comment]
QuickShop 可能在后续重构中添加相关支持。但绝对不是现在。

**Comment by HYDeath** [comment]
@Ghost-chu 这个还有希望兼容folia吗？我不知道这两个库是否有用，一个老外的和国内米饭 bukkit 和 folia 的调度
1: https://github.com/handy-git/FoliaLib
2: https://github.com/Anon8281/UniversalScheduler

**Comment by YuanYuanOwO** [comment]
> @Ghost-chu 这个还有希望兼容folia吗？我不知道这两个库是否有用，一个老外的和国内米饭 bukkit 和 folia 的调度
> 1: https://github.com/handy-git/FoliaLib
> 2: https://github.com/Anon8281/UniversalScheduler

~~Ghostchu将不在负责QuickShop-Hikari的后续开发以及维护工作，关于Folia支持，请关注此PR #1627~~  
最新开发版已支持Folia，欢迎下载测试:)

**Reply by HYDeath** [reply]
> > @Ghost-chu 这个还有希望兼容folia吗？我不知道这两个库是否有用，一个老外的和国内米饭 bukkit 和 folia 的调度
> > 1: https://github.com/handy-git/FoliaLib
> > 2: https://github.com/Anon8281/UniversalScheduler
> 
> Ghostchu将不在负责QuickShop-Hikari的后续开发以及维护工作，关于Folia支持，请关注此PR #1627

好的感谢

**Comment by YuanYuanOwO** [comment]
https://ci.codemc.io/job/Ghost-chu/job/QuickShop-Hikari-SNAPSHOT/lastBuild/artifact/quickshop-bukkit/target/

这里下载开发版
